src/ocl_utils_mod.f90

```diff
@@ -27,6 +28,16 @@ subroutine init_device(device, version_str, context)
     integer(c_intptr_t), allocatable, target :: &
        platform_ids(:), device_ids(:)
     character(len=1,kind=c_char), allocatable, target :: device_name(:)
+    character(len=1) :: strvalue
+
+    ! Set verbosity to false if FORTCL_VERBOSE does not exist (ierr is 1) or
+    ! is equal to "0".
+    CALL get_environment_variable("FORTCL_VERBOSE", strvalue, status=ierr)
+    if (ierr .eq. 1 .or. strvalue .eq. "0") then
```

Comment:
I've just noticed something that I didn't spot last time: Fortran does not support lazy evaluation so in this `if` there's no guarantee that the `strvalue` comparison will be done *after* the check on `ierr`. We need to explicitly check `ierr` first and only then move on to check `strvalue`.

Reply:
tbh I didn't know. But does it matter in this case? It's an 'or' so  both need to be evaluated and there is no memory issue that would make one of them fail is the other is not true. Even if there is a spurious "0" in a uninitialized strvalue the condition still holds.

Reply:
That's all true but I think we should avoid reading from uninitialised locations. Having written that, I realise the simplest solution is simply to initialise strvalue and then we're both happy :-)

Reply:
Done